cmake/Dependencies.cmake

```diff
@@ -95,11 +95,10 @@ if(USE_XPU)
     message(WARNING "Not compiling with XPU. Could NOT find SYCL."
     "Suppress this warning with -DUSE_XPU=OFF.")
     caffe2_update_option(USE_XPU OFF)
-  else()
-    if(LINUX)
-      string(APPEND CMAKE_CXX_FLAGS " -D__INTEL_PREVIEW_BREAKING_CHANGES")
-    endif()
   endif()
+  foreach(flag ${XPU_HOST_CXX_FLAGS})
+    add_definitions(${flag})
+  endforeach()
```

Comment:
This is an anti-pattern, one should not add global flags to all targets, only to the ones that needs them

Reply:
That is definitely a very correct point! I agree with you！
To be honest, I could not find a good method to solve it. For example, `__INTEL_PREVIEW_BREAKING_CHANGES` is a macro used to maintain the ABI BC in <sycl/sycl.hpp>. So we have to propagate this macro to all libraries that will probably use it, such as c10_xpu, libtorch_cpu, libtorch_xpu, libtorch_python, and even third party libraries including oneDNN, torch-xpu-ops, and kineto. I will be going to investigate this issue, and change it when I approach a more elegant method.